### DISCUSSION TITLE
I am confused (URI, Database name)

### DISCUSSION THREAD

**Discussion by Foxomat** [vrtmrz/obsidian-livesync#309] [Q&A]
**TL;DR: where do I get URI and Database name when setting up couchdb locally?
Also, my couchdb seems to be restarting after 2 seconds of uptime every time.**

-----------

I am trying to set up a couchdb server locally.  
So I thought I needed [Setup a CouchDB server](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/setup_own_server.md#setup-a-couchdb-server) and [Quick setup](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/quick_setup.md#quick-setup)

I look into what docker is, get it and get the couchdb image, create the local.ini and launch it via the provided terminal command. I am reminded that I have to "Remember to replace the path with the path to your local.ini", but `COUCHDB_USER=admin` and `COUCHDB_PASSWORD=password` can stay that way? Are these important? I don't know. So I leave them.  
And then? The db seems to be running. Is the next sub-chapter, [Docker Compose](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/setup_own_server.md#docker-compose), the next step, or an alternative to [Docker CLI](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/setup_own_server.md#docker-cli)? I don't know.  

So I decide to proceed with [Quick setup](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/quick_setup.md#quick-setup), but don't get far. I install the plugin in Obsidian and follow the steps, but I can't proceed at [Remote database configuration](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/quick_setup.md#remote-database-configuration-1). Username and password are probably `COUCHDB_USER=admin` and `COUCHDB_PASSWORD=password` from above, but where do I get the URI and database name? I don't know. I try with the values visible in the screenshot, but get an error `could not connect to [...] Remote URI is not valid`. I click on "Check" and absolutely nothing happens.  I take a look at my terminal and search for something that looks like a URL (or URI?). I find `http://any:5984/`, try it, but get an error `could not connect to [...] ERR_NAME_NOT_RESOLVED`.   

Okay, dead end, maybe I get the missing data from [Docker Compose](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/setup_own_server.md#docker-compose)? Back there I create the `docker-compose.yml` and put it in a folder with the `local.ini` (wondering if the root folder `obsidian-livesync` in the example dir hierarchy is supposed to be the one in `.obsidian/plugins`? probably not.)
At the end it says
```
# -d will launch detached so the container runs in background
docker compose up -d
```
Previously, everything that is supposed to be in the terminal was prepended by `$`, so I wonder where I shut put this code snippet, but figure it belongs into the terminal anyway.
That seems to work, `docker ps` shows a running container. But where do I get the URI and Database name?? I still don't know. But the container is restarting all the time. `docker ps` STATUS shows `Up Less than a second` and then `Restarting (1) 15 seconds ago`. What does that mean? Does it need to be fixed? How? I don't know.

So I continue with this https://localhost.run . That gives me some URL like https://[...].lhr.life , which also doesn't work. This time `could not connect to [...] status 503`. Oh and clicking Check now brings up a popup `Checking configuration failed`.

That's it. Idon't know what else to try.

**Comment by vrtmrz** [comment]
Sorry for the very delayed response!
>  but COUCHDB_USER=admin and COUCHDB_PASSWORD=password can stay that way? Are these important? I don't know. So I leave them.

We can configure it after it has been launched! While testing the CouchDB locally, we can leave it.

If you want to check the connectivity on a local computer, the remote URI should be http://localhost:5984, and you can set a database name as you like. From other devices, the URI could be http://192.168.0.x/ (IP address)

The URI via localhost.run looks fine. However, the 503 error means the server has some trouble. As you mentioned restarting, it might caused by that. Could you check the log of the CouchDB?

**Reply by Foxomat** [reply]
I no longer need help because I am now using an alternative method called [Syncthing](https://github.com/syncthing/syncthing).
Thank you anyways for your reply.
